### PR TITLE
[cxx-interop] Add-on commits to previous NS_OPTIONS fix PR

### PR DESCRIPTION
**Pull request by plotfi**
Add-on to address comments from #67036

Addresses some nits, test addons, and some code cleanups/improvement.

### CODE REVIEW

File: lib/ClangImporter/ImportDecl.cpp [hunk 1]
```diff
@@ -816,8 +816,8 @@ static bool isPrintLikeMethod(DeclName name, const DeclContext *dc) {
 using MirroredMethodEntry =
   std::tuple<const clang::ObjCMethodDecl*, ProtocolDecl*, bool /*isAsync*/>;
 
-ImportedType tryImportOptionsTypeForField(const clang::QualType type,
-                                          ClangImporter::Implementation &Impl) {
+ImportedType tryImportOptionsType(clang::QualType type,
```
Comment (zoecarver) [inline lib/ClangImporter/ImportDecl.cpp:819]: Can you say "option set"

Comment (zoecarver) [inline lib/ClangImporter/ImportDecl.cpp:819]:
```suggestion
ImportedType findOptionSetType(clang::QualType type,
```

File: lib/ClangImporter/ImportType.cpp [hunk 2]
```diff
@@ -3250,13 +3250,14 @@ ImportedType ClangImporter::Implementation::importAccessorParamsAndReturnType(
...
+                            : clangDecl->getParamDecl(0)->getType();
+  ImportedType importedNSOptionsType = tryImportOptionsType(fieldType, *this);
 
   // Import the property type, independent of what kind of accessor this is.
-  auto importedType = importPropertyType(property, isFromSystemModule);
-  if (importedNSOptionsType)
-    importedType = importedNSOptionsType;
+  auto importedType = importedNSOptionsType
+                          ? importedNSOptionsType
+                          : importPropertyType(property, isFromSystemModule);
```
Comment (zoecarver) [inline lib/ClangImporter/ImportType.cpp:3260]:
```
ImportedType importedType  = tryImportOptionsType(fieldType, *this);
if (!importedType)
  importedType = importPropertyType(property, isFromSystemModule);
```

**Review by zoecarver — COMMENTED** [review]
Thanks!

File: lib/ClangImporter/ImportDecl.cpp (hunk 1 above)
Comment (plotfi) [reply lib/ClangImporter/ImportDecl.cpp:819]: so `tryImportOptionSetType` ?

Comment (plotfi) [reply lib/ClangImporter/ImportDecl.cpp:819]: Aye, ok so `findOptionSetType` Ok. will do

**Comment by plotfi** [comment]
@swift-ci please test

**Comment by plotfi** [comment]
@swift-ci please test

**Pull request merged**